### PR TITLE
Refactored service to always have defined return value

### PR DESCRIPTION
**Pull request by KonradFak**

### CODE REVIEW

File: src/app/advertiser/advertiser.service.ts [hunk 1]
```diff
@@ -24,11 +26,13 @@ export class AdvertiserService {
...
+  saveCampaign(campaign: Campaign): Observable<Campaign> {
+    return this.http.put(`${environment.apiUrl}/campaign`, { campaign })
+      .map((campaign: Campaign) => campaign);
   }
 
-  getTargetingCriteria() {
-    return this.http.get(`${environment.apiUrl}/campaign_targeting`);
+  getTargetingCriteria(): Observable<TargetingOption[]> {
+    return this.http.get(`${environment.apiUrl}/campaign_targeting`)
+      .map((targetingOption: TargetingOption[]) => targetingOption);
```
Comment (BartoszBilejczyk) [inline src/app/advertiser/advertiser.service.ts]: I would use targetingOptions here as it's an array

File: src/app/publisher/publisher.service.ts [hunk 2]
```diff
@@ -21,12 +22,14 @@ export class PublisherService {
...
+  saveSite(site: Site): Observable<Site> {
+    return this.http.put(`${environment.apiUrl}/site`, { site })
+      .map((site: Site) => site);
   }
 
-  getTargetingCriteria() {
-    return this.http.get(`${environment.apiUrl}/site_targeting`);
+  getTargetingCriteria(): Observable<TargetingOption[]> {
+    return this.http.get(`${environment.apiUrl}/site_targeting`)
+      .map((targetingOption: TargetingOption[]) => targetingOption);
```
Comment (BartoszBilejczyk) [inline src/app/publisher/publisher.service.ts]: The same thing, targetingOptions

File: src/app/settings/settings.service.ts [hunk 3]
```diff
@@ -4,22 +4,24 @@ import { Observable } from 'rxjs/Observable';
...
-  getNotificationsSettings(userId): Observable<any> {
-    return this.http.get(`${environment.apiUrl}/notifications_settings/${userId}`);
+  getNotificationsSettings(userId: number): Observable<NotificationItem[]> {
+    return this.http.get(`${environment.apiUrl}/notifications_settings/${userId}`)
+      .map((notificationSettings: NotificationItem[]) => notificationSettings);
   }
 
-  updateNotificationsSettings(newSettings: object) {
+  updateNotificationsSettings(newSettings: NotificationItem[]) {
     this.http.put(`${environment.apiUrl}/notifications_settings`, newSettings);
```
Comment (BartoszBilejczyk) [inline src/app/settings/settings.service.ts]: In other PUTs you have .map, it's not needed here? Does newSettings type not produce some errors?

Comment (KonradFak) [reply src/app/settings/settings.service.ts]: Here is somewhat different I define type of variable that is parameter of method not what it returns

Comment (BartoszBilejczyk) [reply src/app/settings/settings.service.ts]: aaah true

**Pull request merged**